packages/radix-ui-themes/src/components/checkbox-card-group.tsx

```diff
@@ -0,0 +1,169 @@
+'use client';
+
+import * as React from 'react';
+import classNames from 'classnames';
+import { useDirection } from '@radix-ui/react-direction';
+import { useControllableState } from '@radix-ui/react-use-controllable-state';
+import * as RovingFocusGroup from '@radix-ui/react-roving-focus';
+import { checkboxCardGroupPropDefs } from './checkbox-card-group.props';
+import { extractProps, marginPropDefs } from '../helpers';
+import { Grid } from './grid';
+import { BaseCheckbox } from './base-checkbox';
+
+import type { PropsWithoutRefOrColor, MarginProps, GetPropDefTypes, Responsive } from '../helpers';
+
+type CheckboxCardGroupContextValue = {
+  size?: Responsive<(typeof checkboxCardGroupPropDefs.size.values)[number]>;
+  name?: string;
+  disabled: boolean;
+  required: boolean;
+  value?: string[];
+  onItemCheck(value: string): void;
+  onItemUncheck(value: string): void;
+};
+const CheckboxCardGroupContext = React.createContext<CheckboxCardGroupContextValue>({
+  disabled: false,
+  required: false,
+  onItemCheck() {},
+  onItemUncheck() {},
```

Comment:
I wonder if it's worth using the context package to surface the composition error when the root hasn't been provided? alternatively, if we don't want to faff with the scopes we could guard and throw in a `useContext` hook of our own.

Not sure how important this is, though is does mean the DX differs from some of the other primitives.

Reply:
oh yeah good point about the context scoping too, that's normally done in the primitive, so should do it here for now.